packages/components/src/autocomplete/autocompleter-ui.native.js

```diff
@@ -188,7 +188,9 @@ export function getAutoCompleterUI( autocompleter ) {
 											>
 												<Icon
 													icon={
-														option?.value?.icon?.src
+														option?.value?.icon
+															?.src ||
+														option?.value?.icon
```

Comment:
I see that these lines are highly nested in the file and Prettier are splitting them so I'm wondering if we could have the icon value in a constant instead of here to make it easier to read, something like `const iconSource = option?.value?.icon?.src || option?.value?.icon`, wdyt?

Reply:
I also noticed that we already have a component that handles the block's icon render ([`BlockIcon`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-icon/index.native.js)), I'm not sure if we considered using it to simplify the render function 🤔 .

I did a quick test and looks like it works, the only difference I saw is that in dark mode, the color doesn't match:

_Screenshot taken from iPad Pro, first icon is rendered with `BlockIcon` and the second is with the current implementation:_
<img width="667" alt="Screenshot 2021-08-18 at 16 11 16" src="https://user-images.githubusercontent.com/14905380/129914120-4ba1eadc-0030-46d3-8335-572f6d20d9bb.png">

<img width="676" alt="Screenshot 2021-08-18 at 16 11 05" src="https://user-images.githubusercontent.com/14905380/129914127-dc7f0c53-2b4b-43b5-91cc-5b9724c4c3e1.png">

Reply:
>I also noticed that we already have a component that handles the block's icon render (BlockIcon), I'm not sure if we considered using it to simplify the render function 🤔 .

Thanks for the suggestion! We didn't consider it but I'll add a maintenance task to reuse that component, probably with a few changes so it supports custom styles.

>I see that these lines are highly nested in the file and Prettier are splitting them so I'm wondering if we could have the icon value in a constant instead of here to make it easier to read

Updated 👍